src/tcpinfo_lib.h

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016 measurement-lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Support for fetching tcp measurements through netlink library.
+//
+// This module will include several elements.
+//  Poller - monitors connection data and calls handlers on events.
+//  Parser - parses a single data record into protobuf format.
+//  Logger* - support for writing data records out to log files.
+//        * may be moved to separate module.
+
+#ifndef MLAB_TCPINFO_LIB_H_
+#define MLAB_TCPINFO_LIB_H_
+
+#include "tcpinfo.pb.h"
+
+#include <functional>
+#include <linux/netlink.h>
+
+namespace mlab {
+namespace netlink {
+
+// Key interface elements:
+//  1. Poll all tcp connections, and maintain cache of measurements for each
+//     connection.
+//  2. On connection close, create a mlab.netlink.TCPDiagnosticsProto and
+//     forward it to an arbitrary functor.
+//  3. Create snapshot of all ESTABLISHED connections, and return as
+//     vector<mlab.netlink.TCPDiagnosticsProto>.
+
+// TCPInfoParser parses a single nlmsg struct, and produces an
+// mlab.netlink.TCPDiagnosticsProto.  It is a class, so that we may later extend
+// it to support filtering and other configured behavior.
+class TCPInfoParser {
+ public:
+  // <msg> ownership NOT transfered.
+  TCPDiagnosticsProto ParseNLMsg(const struct nlmsghdr* msg,
+                                 int protocol) const;
+
+  // <msg> ownership NOT transfered.
```

Comment:
Document that proto is an outparam, please. Also, do protobuf enums become C++ enums? If so, please change the type of `protocol` to that. If not, please document that the valid values for `protocol` are contained in the tcpinfo proto.